src/components/BreadcrumbNav.js

```diff
@@ -0,0 +1,70 @@
+import { Link } from 'react-router-dom'
+import Icon from 'material-ui/svg-icons/action/book'
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import Theme from '../../style/theme'
+
+const styles = {
+  container: {
+    fontFamily: Theme.fontFamily,
+    fontVariant: 'all-small-caps',
+    marginBottom: Theme.padding.small,
+    textAlign: 'left'
+  },
+  anchor: {
+    color: Theme.palette.anchorColor,
+    textDecoration: 'none'
+  },
+  icon: {
+    color: Theme.palette.anchorColor,
+    position: 'relative',
+    top: 6,
+    marginRight: 6
+  }
+}
+
+class BreadcrumbNav extends Component {
+  getKey(index) {
+    return `${this.props.keyPrefix}-${index}`
+  }
+
+  render() {
+    const that = this
+
+    function attachKey(item, index) {
```

Comment:
Can you move these functions outside of the render method?

Reply:
The render function gets called many times. It's better not to keep initializing the function again and again.